### DISCUSSION TITLE
Detailed View for relational fields

### DISCUSSION THREAD

**Discussion by kerns** [craftcms/cms#8893] [Ideas]
### Description

Sort of a workaround for [an issue I'm experiencing with PDFs](https://discordapp.com/channels/456442477667418113/456442477667418115/741952946769494057). There are currently only two views for assets - **List** and **Large Thumbnails**. Neither is especially helpful in the context of documents or rich media that cannot be previewed visually.

A 3rd option for asset display called **Detailed** (aka _table view_) could mitigate this limitation while improving the asset management experience across all asset types. This would consist of a table view with filenames, caption fields, file size, etc ...perhaps with icons for popular actions (like _download_ or _preview_) and/or a kebab menu at the end for those or more advanced operations.

![doc_mock](https://user-images.githubusercontent.com/20254/89733014-e60ce680-da52-11ea-821e-bd7edc2c5056.png)

Ideally, it would be extensible with other actions (tagging, sending as email, generating a hashed share link, etc.)

If it was also possible to tab through this table, inline editing asset captions and titles as you went, that would also be extremely powerful relative to the modal state that needs to be opened/closed now, ...but now maybe I'm pushing it.

Also...there is probably already been requested. I got this far without checking. 🙌

**Comment by brandonkelly** [comment]
To be clear, you are specifically talking about Assets _fields_, right?

**Comment by brandonkelly** [comment]
> If it was also possible to tab through this table, inline editing asset captions and titles as you went, that would also be extremely powerful relative to the modal state that needs to be opened/closed now, ...but now maybe I'm pushing it.

That would be taking this too far, I think, because it would likely give the impression that these table values are tied to the current entry, as opposed to globally for the asset.

**Comment by kerns** [comment]
> > If it was also possible to tab through this table, inline editing asset captions and titles as you went, that would also be extremely powerful relative to the modal state that needs to be opened/closed now, ...but now maybe I'm pushing it.
> 
> That would be taking this too far, I think, because it would likely give the impression that these table values are tied to the current entry, as opposed to globally for the asset.

Would it be insane to just allow asset fields to be overridden at the entry level? It would kill two birds because there are so many situations in which a global asset library with stock titles/captions or keywords is useful – but the ability to override those when desired at the entry level (without needing to create a matrix) would be so useful.

**Comment by brandonkelly** [comment]
Are you saying you want to change an asset’s filename  on a per-entry basis (for example)?

**Comment by kerns** [comment]
When you say it like that it sounds crazy. ;) Changes to the filename should always be global ...and probably never editable in a cell you can tab into. And certainly no filename overrides. Please look past my quick mockup.

There are a few issues/ideas I might be blending together unnecessarily here so maybe it's helpful to discuss them in different FRs. 

I'll focus this one on getting a much richer view of asset meta at the entry level, and the desire for a stronger direct link to previewing/editing/downloading assets attached to any given entry _**from the entry_** they are attached to – a logical point of entry.

If you need more info on use cases or issues here I can provide that ...and I'll fold some of these other ideas into a separate FR and they can be dealt with there! :-)

**Comment by brandonkelly** [comment]
Just remembered there is an open issue for introducing relation-specific custom fields – #2502.

**Comment by proimage** [comment]
I'll just add that this might work better as a checkbox than a third option in the View Modes dropdown (that way the table/details view can still show a large thumbnail or a small list). Also, it would be useful to be able to select which custom fields get shown in the details table.
